modules/common/util/funcs_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:object:generate:=true
+
+package util
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGetOr(t *testing.T) {
+
+	tests := []struct {
+		name string
+		data map[string]interface{}
+		key  string
+		want interface{}
+	}{
+		{
+			name: "Key exists with value 111, returns 111",
+			data: map[string]interface{}{"one": "111"},
+			key:  "one",
+			want: "111",
+		},
+		{
+			name: "Key exists and empty string value, returns fallback",
+			data: map[string]interface{}{"one": ""},
+			key:  "one",
+			want: "fallback",
+		},
+		{
+			name: "Key does not exist, returns the fallback",
+			data: map[string]interface{}{"one": "111"},
+			key:  "four",
+			want: "fallback",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			newData := GetOr(tt.data, tt.key, "fallback")
+			g.Expect(newData).To(BeIdenticalTo(tt.want))
+		})
+	}
+}
+
+func TestIsSet(t *testing.T) {
+
+	tests := []struct {
+		name string
+		data map[string]interface{}
+		key  string
+		want interface{}
+	}{
+		{
+			name: "Key exists, returns 111",
+			data: map[string]interface{}{"one": "111"},
+			key:  "one",
+			want: "111",
+		},
+		{
+			name: "Key does not exist, returns false",
+			data: map[string]interface{}{"one": "111"},
+			key:  "four",
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			newData := IsSet(tt.data, tt.key)
+			g.Expect(newData).To(BeIdenticalTo(tt.want))
+		})
+	}
+}
+
+func TestIsJSON(t *testing.T) {
+
+	tests := []struct {
+		name string
+		data string
+		want string
+	}{
+		{
+			name: "Valid json string",
+			data: `{"some":"json"}`,
+			want: "",
```

Comment:
Would be better to include the err nil-ness (word?) too, e.g this would pass if an empty error string is returned

Reply:
done, see bellow